corehq/apps/integration/kyc/services.py

```diff
@@ -1,10 +1,129 @@
+import json
+import os
+import re
+from datetime import datetime
+
+from django.conf import settings
+from django.utils.text import camel_case_to_spaces
+
+import jsonschema
+
+from corehq.apps.hqcase.utils import update_case
 from corehq.apps.integration.kyc.models import UserDataStore
 
 
 class UserCaseNotFound(Exception):
     pass
 
 
+def verify_users(user_objs, config):
+    # TODO: An endpoint to verify a group of users does not seem to be
+    #       available using Chenosis. If we have to do this with
+    #       multiple calls, consider using Celery gevent workers.
+    results = []
+    for user_obj in user_objs:
+        is_verified = verify_user(user_obj, config)
+        save_result(user_obj, config, is_verified)
+        results.append(is_verified)
+    return results
+
+
+def verify_user(user_obj, config):
+    """
+    Verify a user using the Chenosis MTN KYC API.
+
+    Returns True if all user data is accurate above its required
+    threshold, otherwise False.
+    """
+
+    # Documentation: https://apiportal.chenosis.io/en/apis/kyc-verify#endpoints
+    #
+    # Example request:
+    #
+    #     POST https://dev.api.chenosis.io/kycVerify/v1/customers/{customerId}
+    #     {
+    #         "firstName": "John",
+    #         "lastName": "Doe",
+    #         "phoneNumber": "27650600900",
+    #         "emailAddress": "john.doe@email.com",
+    #         "nationalIdNumber": "123456789",
+    #         "streetAddress": "1st Park Avenue, Mzansi, Johannesburg",
+    #         "city": "Johannesburg",
+    #         "postCode": "20200",
+    #         "country": "South Africa"
+    #     }
+    #
+    # Example 200 response:
+    #
+    #     {
+    #         "statusCode": "0000",
+    #         "statusMessage": "Success.",
+    #         "customerId": "27650600900",
+    #         "transactionId": "232TXYZ-212",
+    #         "data": {
+    #             "firstName": 100,
+    #             "lastName": 100,
+    #             "phoneNumber": 100,
+    #             "emailAddress": 100,
+    #             "nationalIdNumber": 100,
+    #             "streetAddress": 100,
+    #             "city": 100,
+    #             "postCode": 100,
+    #             "country": 100
+    #         }
+    #     }
+
+    # TODO: Determine what thresholds we want
+    required_thresholds = {
+        'firstName': 100,
```

Comment:
@kaapstorm How does these scores function?

Reply:
It looks like a percentage of how close the value that was submitted is to the value that MTN has on record. We haven't been able to test this yet, because we don't (yet) have credentials we can use for testing.